lib/styles/components/_buttons.scss

```diff
@@ -56,50 +191,54 @@ $button-hover-alpha: 0.12;
 $button-focus-alpha: 0.24;
 $button-disabled-alpha: 0.6;
 
-$theme-calculated-colors: (
-	'secondary': (
-		'color': $color-secondary,
-		'rgb': $color-secondary-rgb,
-		'hovered': $color-secondary-hovered,
-		'focused': $color-secondary-focused,
-		'disabled': $color-secondary-disabled,
-	),
-	'primary-supporting': (
-		'color': $color-primary-supporting,
-		'rgb': $color-primary-supporting-rgb,
-		'hovered': $color-primary-supporting-hovered,
-		'focused': $color-primary-supporting-focused,
-		'disabled': $color-primary-supporting-disabled,
-	),
-	'secondary-supporting': (
-		'color': $color-secondary-supporting,
-		'rgb': $color-secondary-supporting-rgb,
-		'hovered': $color-secondary-supporting-hovered,
-		'focused': $color-secondary-supporting-focused,
-		'disabled': $color-secondary-supporting-disabled,
-	),
-);
-$regular-colors: (
-	'minor': $color-minor,
-	'minor-supporting': $color-minor-supporting,
-	'content': $color-content,
-	'danger': $color-danger,
-	'danger-supporting': $color-danger-supporting,
-	'warning': $color-warning,
-	'success': $color-success,
-	'inverted': $color-total-white,
-);
-
 .a-button {
 	$self: &;
 
 	@include buttonM;
+	@include setButtonColor(map-get($button-colors, 'primary', 'filled', 'color'));
+	@include setButtonBackground(
+		map-get($button-colors, 'primary', 'filled', 'background'),
+		map-get($button-colors, 'primary', 'filled', 'background-hovered'),
+		map-get($button-colors, 'primary', 'filled', 'background-focused'),
+		map-get($button-colors, 'primary', 'filled', 'background-disabled')
+	);
+	@include setButtonDisabled(
+		map-get($button-colors, 'primary', 'filled', 'disabled', 'color'),
+		map-get($button-colors, 'primary', 'filled', 'disabled', 'icon'),
+		map-get($button-colors, 'primary', 'filled', 'disabled', 'background')
+	);
+	@include setButtonAdditions(
+		map-get($button-colors, 'primary', 'filled', 'ripple'),
+		'',
+		map-get($button-colors, 'primary', 'filled', 'icon')
+	);
+
+	@each $color-name, $color-map in $icon-button-colors {
```

Comment:
Now, I see that `primary` is defined two times. Can we exclude it from the loop?

Reply:
Generally I'm removing default styles because we cant use component without color from the list. So situation when color will be not defined will not occur.